tests/datasets/test_transfer.py

```diff
@@ -127,29 +134,36 @@ def test_chunking_and_labelling_data_of_exact_sizes(mocker):
 
 
 def test_chunking_and_labelling_of_smaller_sizes(mocker):
-    mocker.patch("faculty.datasets.transfer.UPLOAD_CHUNK_SIZE", 12)
+    chunk_size = 12
     content = [b"1111", b"2222", b"last"]
-    chunks = transfer._rechunk_and_label_as_last(content)
+    chunks = transfer._rechunk_and_label_as_last(content, chunk_size)
     assert list(chunks) == [(b"11112222last", True)]
 
 
 def test_chunking_and_labelling_of_greater_sizes(mocker):
-    mocker.patch("faculty.datasets.transfer.UPLOAD_CHUNK_SIZE", 4)
+    chunk_size = 4
     content = [b"11112222last"]
-    chunks = transfer._rechunk_and_label_as_last(content)
+    chunks = transfer._rechunk_and_label_as_last(content, chunk_size)
     assert list(chunks) == [
         (b"1111", False),
         (b"2222", False),
         (b"last", True),
     ]
 
 
+def test_s3_dynamic_chunk(mocker):
```

Comment:
It's not clear which function (`_s3_chunk_size` or `_rechunk_data`) is really being tested here. I'd prefer that only the  function under test gets called.